### PR TITLE
Clarifications about beforeMarshal

### PR DESCRIPTION
**Pull request by lorenzo**
This should help people trying to understand beforeMarshal

### CODE REVIEW

File: en/orm/saving-data.rst [hunk 1]
```diff
@@ -561,12 +561,32 @@ request data just before entities are created::
     // In a table or behavior class
     public function beforeMarshal(Event $event, ArrayObject $data, ArrayObject $options)
     {
-        $data['username'] .= 'user';
+       if (isset($data['username'])) {
+           $data['username'] = strtolower($data['username']);
```
Comment (dereuromark) [inline en/orm/saving-data.rst]: needs to be mb_ :)

File: en/orm/saving-data.rst [hunk 2]
```diff
@@ -561,12 +561,32 @@ request data just before entities are created::
...
+    public function beforeMarshal(Event $event, ArrayObject $data, ArrayObject $options)
+    {
+        foreach ($data as $key => $value) {
+            if (is_string($value)) {
+                $data[$key] = trim($value);
+            }
+        }
+    }
+
+Becuase of how the marshalling process works, if a field does not pass validation it will automatically removed from the data array and not be copied into the entity. This is to prevent having inconsistent data in the entity object.
```
Comment (dereuromark) [inline en/orm/saving-data.rst]: typo because

**Comment by Xety** [comment]
:+1: Thanks, now i understand the real interest of the `beforeMarshal`.

File: en/orm/saving-data.rst [hunk 3]
```diff
@@ -561,12 +561,32 @@ request data just before entities are created::
...
+    public function beforeMarshal(Event $event, ArrayObject $data, ArrayObject $options)
+    {
+        foreach ($data as $key => $value) {
+            if (is_string($value)) {
+                $data[$key] = trim($value);
+            }
+        }
+    }
+
+Because of how the marshalling process works, if a field does not pass validation it will automatically removed from the data array and not be copied into the entity. This is to prevent having inconsistent data in the entity object.
```
Comment (antograssiot) [inline en/orm/saving-data.rst]: ``it will automatically removed`` -> ``it will automatically be removed`` ?

**Comment by antograssiot** [comment]
I wrapped the long lines and added the french translation (cc: @Xety @cake17)

File: fr/orm/saving-data.rst [hunk 4]
```diff
@@ -595,6 +595,34 @@ entities ne soient créées::
 Le paramètre ``$data`` est une instance ``ArrayObject``, donc vous n'avez pas
 à la retourner pour changer les données utilisées pour créer les entities.
 
+Le but principal de ``beforeMarshal`` est d'aider les utilisateurs à passer
+le process de validation lorsque des erreurs simples peuvent être résolues
+automatiquement, ou lorsque les données doivent être restructurées pour être
+mise dans les bons champs.
+
+L'event ``beforMarshal`` est lancé juste au début du process de validation.
+Une des raisons à cela est que ``beforeMarshal`` est autoridé à modifier les
```
Comment (HavokInspiration) [inline fr/orm/saving-data.rst]: autorisé*

File: fr/orm/saving-data.rst [hunk 5]
```diff
@@ -595,6 +595,34 @@ entities ne soient créées::
...
+règles de validation et les options d'enregistrment, telles que la whitelist
```
Comment (HavokInspiration) [inline fr/orm/saving-data.rst]: enregistrement*

File: fr/orm/saving-data.rst [hunk 6]
```diff
@@ -595,6 +595,34 @@ entities ne soient créées::
...
+        foreach ($data as $key => $value) {
+            if (is_string($value)) {
+                $data[$key] = trim($value);
+            }
+        }
+    }
+
+A cause de la manière dont le process de marshalling fonctionne, si un champ ne
+passe pas la validation, il sera automatiquement supprimé du tableau de données
+et ne sera pas copié dans l'entité. cela évite d'avoir des données incohérentes
```
Comment (HavokInspiration) [inline fr/orm/saving-data.rst]: dans **l'entity** plutôt ?

**Comment by HavokInspiration** [comment]
Translation looks good to me besides the few typos I noticed.
Nice work @antograssiot :+1:

**Comment by markstory** [comment]
Fixed more typos.

**Comment by cake17** [comment]
@antograssiot nice work

**Pull request merged**